File: TP01/Palindromo_Q1&2/Palindromo.java
 class Palindromo {

   public static boolean isFim(String s){
      return (s.length() >= 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
   }

   public static boolean IsPalindromo (String s){
      boolean resp = true;

      for(int i = 0; i < s.length()/2; i++){
         if(s.charAt(i) != s.charAt(s.length()-1-i)){
            resp = false;
         }
      }
      return resp;
   }

   public static void main (String[] args){
      String[] entrada = new String[1000];
      int numEntrada = 0;

      //Leitura da entrada padrao
      do {
         entrada[numEntrada] = MyIO.readLine();
      } while (isFim(entrada[numEntrada++]) == false);
      numEntrada--;   //Desconsiderar ultima linha contendo a palavra FIM

      //Para cada linha de entrada, gerando uma de saida contendo o numero de letras maiusculas da entrada
      for(int i = 0; i < numEntrada; i++){
         MyIO.println(IsPalindromo(entrada[i]) ? "SIM" : "NAO");
      }
   }
}

File: TP01/Palindromo_Q1&2/Palindromo.cpp
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h> 

   bool Fim(char* palavra){
      return (strlen(palavra) >= 3 && palavra[0] == 'F' && palavra[1] == 'I' && palavra[2] == 'M');
   }

   bool IsPalindromo (char* s){
      bool resp = true;

      for(int i = 0; i < strlen(s)/2; i++){
         if(s[i] != s[strlen(s)-2-i]){
            resp = false;
         }
      }
      return resp;
   }

int main(int argc, char** argv){

    char palavra[1000][1000];
    int numEntrada = 0;
    
   //Ler da entrada padrao as palavras e chamar mestodo para testar se eh a palavra FIM
    do {
       fgets(palavra[numEntrada], 1000, stdin);
    } while (Fim(palavra[numEntrada++]) == false);
    numEntrada--;   
  
   //Chamar metodo de contar letras maisculas e depois exibir na tela a quantidade de letras maisculas.
    for(int i = 0; i < numEntrada; i++){
       IsPalindromo(palavra[i]) ? printf("SIM\n") : printf("NAO\n");
    }
}